frontend/public/components/utils/webhooks.tsx

```diff
@@ -56,17 +56,31 @@ export const WebhookTriggers: React.FC<WebhookTriggersProps> = (props) => {
   });
   const tableColumnClasses = getTableColumnClasses(canGetSecret);
   const [webhookSecrets, setWebhookSecrets] = React.useState<K8sResourceKind[]>([]);
+  const [webhookTriggers, setWebhookTriggers] = React.useState<WebhookTrigger[]>([]);
+  const [secretNames, setSecretNames] = React.useState<string[]>([]);
   const [secretErrors, setSecretErrors] = React.useState<string[]>([]);
   const [isLoaded, setLoaded] = React.useState(false);
 
-  const webhooks: WebhookTrigger[] = _.filter(triggers, ({ type }) => webhookTriggers.has(type));
-  const secretNames: string[] = _.uniq(
-    webhooks.reduce((acc: string[], webhook: WebhookTrigger): string[] => {
-      const triggerProperty = getTriggerProperty(webhook);
-      const secretName = _.get(webhook, [triggerProperty, 'secretReference', 'name']);
-      return secretName ? [...acc, secretName] : acc;
-    }, []),
-  );
+  React.useEffect(() => {
+    const newWebhookTriggers: WebhookTrigger[] = _.filter(triggers, ({ type }) =>
+      webhookTriggerTypes.has(type),
+    );
+    if (_.isEqual(newWebhookTriggers, webhookTriggers)) {
+      return;
+    }
+    setWebhookTriggers(newWebhookTriggers);
+  }, [triggers, webhookTriggers]);
```

Comment:
Will adding webhookTriggers as a dep cause issues here since it is updated within this effect?

Reply:
It should not since the `webhookTriggers` will be updated only when the webhookTriggers are update(added/removed)